### ISSUE TITLE
Handle error when create allocation

### ISSUE THREAD

**Issue by haule24hdev** [0chain/0chain#498]
![Screenshot from 2021-08-24 10-24-02](https://user-images.githubusercontent.com/50743635/130550846-6c8a7296-2fa5-499d-85d9-6ddac263fda3.png)
![Screenshot from 2021-08-24 10-25-18](https://user-images.githubusercontent.com/50743635/130550949-adf464fe-f6e1-4b83-825e-4e0e0f408e01.png)
This is curl:
curl 'https://prom.0chain.net/sharder02/v1/screst/6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7/allocation_min_lock?allocation_data=%7B%22data_shards%22:4,%22parity_shards%22:2,%22owner_id%22:%22ba71cfe98f9ef8745afec340d2c3a65be2559e57a0d0390fe8070ca2d234a0a1%22,%22owner_public_key%22:%22149fb08b6d5a7ad08b5c0fad1bd08a0c724131e5ec5dd8c19661920500d3b0168a481500f6bb8ee4a37f8da4f1f3a2c015b16d9ee7ccbeb35bc1c909ec99ce8b%22,%22size%22:2147483648,%22expiration_date%22:1632367309,%22read_price_range%22:%7B%22min%22:177000000,%22max%22:273084944%7D,%22write_price_range%22:%7B%22min%22:173084944,%22max%22:273084944%7D,%22max_challenge_completion_time%22:3600000000000,%22preferred_blobbers%22:null%7D' \
  -H 'authority: prom.0chain.net' \
  -H 'sec-ch-ua: "Chromium";v="92", " Not A;Brand";v="99", "Google Chrome";v="92"' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36' \
  -H 'sec-fetch-site: same-origin' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-dest: empty' \
  -H 'referer: https://prom.0chain.net/0box/dashboard/allocateStorage' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'cookie: __stripe_mid=56ed0203-cf50-49b8-a1ab-32e95e26795ba137b0; __stripe_sid=0f89996d-d62d-4fa7-af29-8f7fbd4c6964412d71' \
  --compressed

**Comment by shaktals** [comment]
Maybe a better/more specific name for this issue title would be something like:
`Error trying to get blobber list to honor new advanced allocation`

Btw, I tried to find docs for the `allocation_min_lock` endpoint, but couldn't find any.

**Comment by shaktals** [comment]
This is also is hindering the resolution of [another issue](https://github.com/0chain/zboxcli/issues/94#issuecomment-904200281) in `zboxcli`.

**Comment by Sriep** [comment]
The error means that there are not enough blobbers that satisfy the parameters of the allocation request.

Ether add more blobbers or reduce parameters like size, shards, expiration time etc.

**Comment by shaktals** [comment]
@Sriep I played with it again. and I found out the bug happens only when we move the lower boundary of prices. When you move the upper boundary the error goes away. Now it looks like a _front end_ error indeed. @haule24hdev will you please confirm?

**Comment by krishnemd** [comment]
Here the error reporting is fine as it says that given parameters do not match with any of the blobber specification. This error will occur even in Simple mode of creating allocation when unnatural size is entered in the form. Backend wise error reporting is fine so let's close the issue.

**Comment by shaktals** [comment]
Just one thing though, it seems like when the parameters can not be honored by any blobber, the backend response code is `500 - internal server error`. That is not a semantically correct error code, imho it should be fixed.

My preference would be a `200` or `204` code with empty response, stating that the request was successfully made to a valid route, but there are no content.

A `404` always reminds me of non-existing resource/route.

**Comment by krishnemd** [comment]
> Just one thing though, it seems like when the parameters can not be honored by any blobber, the backend response code is `500 - internal server error`. That is not a semantically correct error code, imho it should be fixed.
> 
> My preference would be a `200` or `204` code with empty response, stating that the request was successfully made to a valid route, but there are no content.
> 
> A `404` always reminds me of non-existing resource/route.

Yes in that case its an issue. I can see it being raised in multiple repos.
https://github.com/0chain/zboxcli/issues/53

**Comment by shaktals** [comment]
Cool, I'll close this and open a new specific issue regarding that.
